src/engine/Server.h

```diff
@@ -156,9 +156,15 @@ class Server {
 
   /// Run the SPARQL parser and then the query planner on the `query`. All
   /// computation is performed on the `threadPool_`.
-  net::awaitable<PlannedQuery> parseAndPlan(
-      const std::string& query, QueryExecutionContext& qec,
-      SharedCancellationHandle handle) const;
+  net::awaitable<PlannedQuery> parseAndPlan(const std::string& query,
+                                            QueryExecutionContext& qec,
+                                            SharedCancellationHandle handle,
+                                            TimeLimit timeLimit) const;
+
+  ///
```

Comment:
Should tell us what this returns. 
If you create a local templated struct, you can even make this part of the interface:
```
template <SomeInvocableConceptThatTellsHowToUseCallback Callback>
struct CancellationHandleAndCallback {
  CancellationHandle handle_;
  // The callback can be used to....
  Callback callback_;
};
ad_utility::InstantiationOf<CancellationHandleAndCallback> auto setup.....(...) const.
```

(The return type is `auto` but enforce statically that it always fulfills a given concept.

Reply:
(In general a small comment is nice).